airflow/contrib/operators/gcp_vision_operator.py

```diff
@@ -0,0 +1,603 @@
+    """
+    Gets information associated with a `Product`.
+
+    Possible errors:
+
+    - Returns NOT_FOUND if the `Product` does not exist.
+
+    :param location: (Required) The region where the Product is located. Valid regions are:
+        us-east1, us-west1, europe-west1, asia-east1
+    :type location: str
+    :param product_id: (Required) The resource id of this Product.
+    :type product_id: str
+    :param project_id: (Optional) The project in which the Product is located. If set to
+        None or missing, the default project_id from the GCP connection is used.
+    :type project_id: str
+    :param retry: (Optional) A retry object used to retry requests. If `None` is
+        specified, requests will not be retried.
+    :type retry: google.api_core.retry.Retry
+        https://googleapis.github.io/google-cloud-python/latest/core/retry.html#google.api_core.retry.Retry
+    :param timeout: (Optional) The amount of time, in seconds, to wait for the request to
+        complete. Note that if retry is specified, the timeout applies to each individual
+        attempt.
+    :type timeout: float
+    :param metadata: (Optional) Additional metadata that is provided to the method.
+    :type metadata: Sequence[Tuple[str, str]]
+    :param gcp_conn_id: The connection ID used to connect to Google Cloud Platform.
+    :type gcp_conn_id: str
+
+    """
+    # [START vision_product_get_template_fields]
+    template_fields = ('location', 'project_id', 'product_id', 'gcp_conn_id')
+    # [END vision_product_get_template_fields]
+
+    @apply_defaults
+    def __init__(self,
+                 location,
+                 product_id,
+                 project_id=None,
+                 retry=None,
+                 timeout=None,
+                 metadata=None,
+                 gcp_conn_id='google_cloud_default',
+                 *args, **kwargs):
+        super(CloudVisionProductGetOperator, self).__init__(*args, **kwargs)
+        self.location = location
+        self.product_id = product_id
+        self.project_id = project_id
+        self.retry = retry
+        self.timeout = timeout
+        self.metadata = metadata
+        self.gcp_conn_id = gcp_conn_id
+        self._hook = CloudVisionHook(gcp_conn_id=self.gcp_conn_id)
+
+    def execute(self, context):
+        return self._hook.get_product(location=self.location,
+                                      product_id=self.product_id,
+                                      project_id=self.project_id,
+                                      retry=self.retry,
+                                      timeout=self.timeout,
+                                      metadata=self.metadata)
+
+
+class CloudVisionProductUpdateOperator(BaseOperator):
+    """
+    Makes changes to a Product resource. Only the display_name, description, and labels fields can be
+    updated right now.
+
+    If labels are updated, the change will not be reflected in queries until the next index time.
+
+    Note:
```

Comment:
```suggestion
    .. note:: To locate the `Product` resource, its `name` in the form...........
```
See example: https://raw.githubusercontent.com/apache/airflow/master/docs/howto/initialize-database.rst

Reply:
Done